### ISSUE TITLE
Terminal improvements

### ISSUE THREAD

**Issue by mattpass** [mattpass/ICEcoder#668]
The terminal plugin has been remade and now part of ICEcoder's core but there are still a few improvements that could be made:

- [ ] Correct path to be shown in white on sending command
- [ ] Currently saves blank in history if you submit a blank command, stop that
- [ ] Add dotfile aliases from https://github.com/mattpass/dotfiles
- [ ] Remove session for cwd - use JS var & send that value with command over XHR
- [ ] Right click a file/dir to open path in terminal
- [ ] SSH able using a lib such as phpseclib, Xwiterm (https://code.google.com/p/xwiterm/) or PHP Shell (http://phpshell.sourceforge.net/)
- [ ] Autocomplete with tab key
- [ ] True SSH access via the browser (as the php 'sandbox' can be somewhat limiting. User $PATH not being loaded, etc)

**Comment by aggsol** [comment]
I am especially interestedin the "true"SSH access or a more bash-like behaviour. Would you recommend installing an extra shell next to ICEcode? To speed thing up, I am willing to set a bounty. I am sure I am not the only one missing a proper shell.

**Comment by mattpass** [comment]
@aggsol Yes, it's only a basic shell at the moment and  UX/UI not quite useful as a real terminal just yet. It's also restricted by what PHP is permitted to do under your wwwuser.

If you're looking for a full powered shell, checkout Butterfly Terminal - https://github.com/paradoxxxzero/butterfly - it's Python based but probably covers what you're looking for?

**Comment by aggsol** [comment]
@mattpass ah thank you, I will have a look :+1:
